test/e2e/config/docker.yaml

```diff
@@ -115,6 +115,8 @@ variables:
   COREDNS_VERSION_UPGRADE_TO: "1.8.4"
   KUBERNETES_VERSION_UPGRADE_TO: "v1.22.0"
   KUBERNETES_VERSION_UPGRADE_FROM: "v1.21.2"
+  # CAPI v0.3 cannot be deployed on Kubernetes >= v1.22.
```

Comment:
Might be I'm wrong but the problem here is CAPI v0.3 not able to create workload cluster >= v1.22. right?

Reply:
It can create workload clusters, it's the management cluster that's blocked on 1.21

Reply:
Exactly. We're not able to deploy CAPI v0.3 on top of a Kubernetes v1.22 cluster. In case of this e2e test the CAPI v0.3 deployment gets stuck because cert-manager 1.1 does not come up on the v1.22 cluster.

Reply:
Ok, I got confused because we are using the new variable both for the management cluster (I initially missed this) and for the workload cluster.

Might be it could help to call the new variable INIT_WITH_KUBERNETES_VERSION and move it down near to INIT_WITH_BINARY (That sets the CAPI version to start from in clusterctl upgrades)

Also, let's make sure to use this variable only for the management cluster

Reply:
Done. With the old const name it might have been confusing to use two different Kubernetes versions, but not it makes more sense.

Of course this will only work as long as we support new Kubernetes versions with the v0.3.x CAPI version. But it's nice that we also test CAPI v0.3.x against "latest stable" Kubernets with this test.